### PR TITLE
Feat/004 llist

### PR DESCRIPTION
**Pull request by salvadorz**
# Description

Adding Linked list library. The list can hold and native or custom datatype. It offers MACROS as API to handle a better usage and ensure freeing the memory.

# Validation

A Unit Test it's created where test all MACROS and APIs from the library.
image

> No Memory leak is present

- [X] test_llist_lib
- [X] 90% Branches on Code Coverage - missing covering the error handling  and the delete
- 
image

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/lib/llist/test_llist.c [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  TEST_ASSERT_EQUAL_VAL_MSG(MAX_LLIST_LEN, llist_get_size(my_struct_list),
+                            "llist shall be holding MAX_LLIST_LEN (10) elements");
+  llist_print(my_struct_list, print_my_struct);
+
+  /** Now using POP_DATA */
+  for (uint32_t i = 0; i < MAX_LLIST_LEN; ++i) {
+    my_struct_t ret = LLIST_POP_DATA(my_struct_list);
+    TEST_ASSERT_EQUAL_VAL_MSG(i + 1, ret.data, "Increasing values...");
+    TEST_ASSERT_EQUAL_VAL_MSG((MAX_LLIST_LEN - 1) - i, llist_get_size(my_struct_list),
+                              "llist shall be decreasing it's elements");
```
Comment (salvadorz) [inline tests/lib/llist/test_llist.c:149]: Add a test with  fills and empty the list with `llist_delete_list(&head);`

**Review by salvadorz — COMMENTED** [review]
Proposal: Add another test case to exercise the llist_delete()
